### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Ilya-brinchuk**

### CODE REVIEW

File: src/main/java/core/basesyntax/model/Car.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    private long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer, List<Driver> drivers) {
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = drivers;
+    }
```
Comment (RuslanIbIsh) [inline src/main/java/core/basesyntax/model/Car.java:48]: в Entity желательно добавить getters and setters

**Review by RuslanIbIsh — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -9,8 +9,59 @@
...
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
     </properties>
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (RomanDubovskyi) [inline pom.xml]: let's not add dependencies that we don't need now, it's a bad practice

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        for (Manufacturer manufacturer : Storage.manufacturers) {
+            if (manufacturer.getId() == id) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: what is the right way to compare two objects ? 
it's better to use streams here, will make the code shorter

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        for (Manufacturer oldManufacturer : Storage.manufacturers) {
+            if (oldManufacturer.getId() == manufacturer.getId()) {
+                int index = Storage.manufacturers.indexOf(oldManufacturer);
+                Storage.manufacturers.set(index, manufacturer);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: you may get IndexOutOfBound here which will not be obvious for the user of your code, I suggest this approach 
IntStream.range(0, Storage.manufacturers.size())
                .filter(i -> Storage.manufacturers.get(i).getId().equals(manufacturer.getId()))
                .forEach(i -> Storage.manufacturers.set(i, manufacturer));
        return manufacturer;

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                return manufacturer;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        for (Manufacturer manufacturer : Storage.manufacturers) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: try to use method removeIf here

File: src/main/java/core/basesyntax/model/Car.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Car {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/model/Car.java:6]: add getters and setters  so we will have ready to use model

File: src/main/java/core/basesyntax/model/Driver.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.model;
+
+public class Driver {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/model/Driver.java:3]: same here

File: src/main/java/core/basesyntax/service/impl/ManufacturerServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import core.basesyntax.lib.Inject;
+import core.basesyntax.lib.Service;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.service.ManufacturerService;
+import java.util.List;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    ManufacturerDao manufacturerDao;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/service/impl/ManufacturerServiceImpl.java]: what about access modifiers ?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        IntStream.range(0, Storage.manufacturers.size())
+                .filter(i -> Storage.manufacturers.get(i).getId() == manufacturer.getId())
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: is == correct?

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+                .forEach(i -> Storage.manufacturers.set(i, manufacturer));
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.manufacturers.removeIf(manufacturer -> manufacturer.getId() == id);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: same

File: src/main/java/core/basesyntax/service/impl/ManufacturerServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    private ManufacturerDao manufacturerDao;
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return manufacturerDao.create(manufacturer);
+
+    }
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/service/impl/ManufacturerServiceImpl.java]:
```suggestion
        return manufacturerDao.create(manufacturer);
    }
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
almost there

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**